features/0685-pickup-v2/README.md

```diff
@@ -0,0 +1,184 @@
+# 0212: Pickup Protocol 2.0
+
+- Authors: [Sam Curren](telegramsam@gmail.com)
+- Status: [PROPOSED](/README.md#proposed)
+- Since: 2019-09-03
+- Status Note: Initial version
+- Start Date: 2020-12-22
+- Tags: [feature](/tags.md#feature), [protocol](/tags.md#protocol)
+
+## Summary
+
+A protocol to facilitate an agent picking up messages held at a mediator.
+
+## Motivation
+
+Messages can be picked up simply by sending a message to the _Mediator_ with a `return_route` decorator specified. This mechanism is implicit, and lacks some desired behavior made possible by more explicit messages.
+
+This protocol is the explicit companion to the implicit method of picking up messages.
+
+## Tutorial
+
+### Roles
+
+**Mediator** - The agent that has messages waiting for pickup by the _recipient_.
+**Recipient** - The agent who is picking up messages.
+
+### Flow
+
+status can be used to see how many messages are pending. The Recipient sends a StatusRequest message to the Mediator.
+
+Messages are retrieved when the Recipient  sends a `delivery_request` message to the Mediator.
+
+Message delivery is confirmed with a `message_received` acknowledgement, at which point the message can be removed from the mediator's queue.
+
+When live delivery is enabled, messages that arrive when an existing connection exists are delivered over the connection immediately.
+
+
+
+## Reference
+
+Each message sent should use the ~transport decorator as follows. This has been ommitted from the examples for brevity.
+
+```json=
+"~transport": {
+    turn_route": "thread"
+}
```

Comment:
return_route thread requires the `return_route_thread` parameter to be present. 
```suggestion
"~transport": {
    "return_route": "thread",
    "return_route_thread": "<thread-uuid>"
}
```

Reply:
Is there a specific reason you choose thread? I think this should be all, because the mediator can't know the thread?

Reply:
I have mixed feelings about this because of the changes in return_route coming in DIDComm V2. Specifically, using thread was designed to allow the remote agent to communicate with the mediator without invoking a flood of mediated messages. By going the explicit way we are going, including live mode, this flow isn't needed at all.

Given that thinking, I'm going to switch it to all.